### ISSUE TITLE
Mac Doc - Updating the OS doesn't automatically update xcode

### ISSUE THREAD

**Issue by jwokaty** [Bioconductor/BBS#237]

**Comment by jwokaty** [comment]
Additionally, you must re-run

```
Installation problem: before you can use the /Users/biocbuild/BBS/utils/macosx-make-tgz.sh script, you must
install chown-rootadmin in the same directory as /Users/biocbuild/BBS/utils/macosx-make-tgz.sh like this:
  cd /Users/biocbuild/BBS/utils
  gcc chown-rootadmin.c -o chown-rootadmin
  sudo chown root:admin chown-rootadmin
  sudo chmod 4750 chown-rootadmin
```

**Comment by hpages** [comment]
Note that `chown-rootadmin.c` is part of the BBS source. I wonder what recompilation of `chown-rootadmin.c` has to do with Xcode and OS updates :thinking:

**Comment by jwokaty** [comment]
Yes, me too. However, it's why we haven't seen any build report because all the packages were erroring on buildbin.

**Comment by hpages** [comment]
Yes, maybe. But again, I wonder how this could have anything to do with Xcode and OS updates. AFAIK an OS update would never delete user files. It's good that you caught this problem on kjohnson but it's unlikely to be related to the topic of this issue. Only mentioning this for the sake of keeping issues organized and focused.

**Comment by hpages** [comment]
It seems that kjohnson actually got upgraded from Monterey to Ventura:
```
kjohnson:~ biocbuild$ uname -a
Darwin kjohnson 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:30 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T8103 arm64
```
This is no ordinary system update, more like a major OS upgrade. Was this intended?

macOS Ventura was released only 2 weeks ago. Sounds risky to me to go for this upgrade so early. The timing is not great either, given that kjohnson was participating to the BioC 3.16 release. Stability is very important in release times.

This might explain why you had to take special action to get a new Xcode (still doesn't explain what happened to `chown-rootadmin` though).

In any case `nodes/nodespecs.py` would need to be updated as documented [here](https://github.com/Bioconductor/BBS/blob/master/Doc/Prepare-macOS-Mojave-HOWTO.md#13-apply-all-software-updates).

**Comment by jwokaty** [comment]
Yes, I updated to Ventura intentionally and it is likely why I had to update Xcode, but stability is a good point.

**Comment by hpages** [comment]
Not just stability. Other things to take into account are:
- Time: How much time you're willing to spend on troubleshooting the issues that a major OS upgrade like that is likely to introduce. We have so many things to deal with already during a release, that we must be careful to not introduce changes that are likely to break things, especially when those changes are not urgent and might require time to troubleshoot.
- Developer expectations: We put a lot of pressure on Bioconductor developers in the last weeks before a release to make sure that they fix build issues, and most of them work hard to achieve that. But there's a risk that this OS upgrade will cause new packages to fail. For example the upgrade is likely to replace the compilers with newer versions that have never been used before to build Bioconductor packages. Some of the new build failures we see could be for packages that their developers worked hard to get ready for the BioC 3.16 release. This is the reason why we didn't upgrade nebbiolo2 from Ubuntu 20.04 to Ubuntu 22.04 for this release.
- Reproducibility: Most developers that are on a Mac are probably going to stick to Big Sur or Monterey for a while (probably for good reasons), which will make it difficult for them to reproduce/troubleshoot the build failures they see on the build report.
